File: src/Application Specific/Classes/Entity Collections/Van/VanValidator.vb
Namespace Entity

Namespace Vans

Public Class VanValidator
    'make sure that contact object is valid
    Inherits BaseValidator

    Public Sub Validate(ByVal oVan As Van)

        'make sure that contact object is valid
        If oVan.Errors.Count > 0 Then
            Dim de As DictionaryEntry
            For Each de In oVan.Errors
                Me.AddCriticalMessage(CStr(de.Value))
            Next
        End If


                If oVan.Registration.Trim.Length = 0 Then
                    Me.AddCriticalMessage("Profile Name Missing")
                Else
                    If oVan.Registration.Contains("*") Then
                        Me.AddCriticalMessage("An asterix (*) cannot be placed in the profile name")
                    Else
                        If DB.Van.Check_Unique_Registration(oVan.Registration, oVan.VanID) = False Then
                            Me.AddCriticalMessage("Profile name already exists")
                        End If
                    End If
                End If

                If Me.ValidatorMessages.CriticalMessages.Count > 0 Then
                    Throw New ValidationException(Me)
                End If

    End Sub

End Class

End Namespace

End Namespace

File: src/Application Specific/Classes/Entity Collections/Van/VanValidator.cs
using System.Collections;
using Microsoft.VisualBasic.CompilerServices;

namespace FSM.Entity
{
    namespace Vans
    {
        // make sure that contact object is valid
        public class VanValidator : BaseValidator
        {
            public void Validate(Van oVan)
            {
                // make sure that contact object is valid
                if (oVan.Errors.Count > 0)
                {
                    foreach (DictionaryEntry de in oVan.Errors)
                        AddCriticalMessage(Conversions.ToString(de.Value));
                }

                if (oVan.Registration.Trim().Length == 0)
                {
                    AddCriticalMessage("Profile Name Missing");
                }
                else if (oVan.Registration.Contains("*"))
                {
                    AddCriticalMessage("An asterix (*) cannot be placed in the profile name");
                }
                else if (App.DB.Van.Check_Unique_Registration(oVan.Registration, oVan.VanID) == false)
                {
                    AddCriticalMessage("Profile name already exists");
                }

                if (ValidatorMessages.CriticalMessages.Count > 0)
                {
                    throw new ValidationException(this);
                }
            }
        }
    }
}